zerver/tests/test_events.py

```diff
@@ -1861,10 +1861,19 @@ def test_change_is_admin(self) -> None:
         self.user_profile.refresh_from_db()
 
         do_change_user_role(self.user_profile, UserProfile.ROLE_MEMBER, acting_user=None)
+
+        self.make_stream("Test private stream", invite_only=True)
+        self.subscribe(self.example_user("othello"), "Test private stream")
+
         for role in [UserProfile.ROLE_REALM_ADMINISTRATOR, UserProfile.ROLE_MEMBER]:
+            if role == UserProfile.ROLE_REALM_ADMINISTRATOR:
+                num_events = 6
+            else:
+                num_events = 5
```

Comment:
Yeah this is really complicated. I think the refactor that we'll probably want to do is assemble a list of expected (event, validator) pairs and then compute `num_events` from that, such that we  only have one list of conditionals.

Reply:
Yeah, we could do something like that. Noted it and will look into this.